demos/faults.h

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under both the 3-Clause BSD License and the GPLv2, found in the
+ * LICENSE and LICENSE.GPL-2.0 files, respectively, in the root directory.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause OR GPL-2.0
+ */
+
+#ifndef DEMOS_FAULTS_H_
+#define DEMOS_FAULTS_H_
+
+#include <signal.h>
+
+#include <functional>
+
+// See "The siginfo_t argument to a SA_SIGINFO handler"
+// at https://www.man7.org/linux/man-pages/man2/sigaction.2.html
+using PosixFaultHandler = std::function<void(int, siginfo_t*, void*)>;
```

Comment:
This doesn't seem to be used yet except with an empty callback. Will we *need* it, or can callers always check the return value instead?

Signal handlers are super scary / hard to do right thing (signal safety is high impossible), so I'd prefer if we didn't expose it at all, if at all possible.

And here's a case in point: if we do want to support it, we must not use `std::function`, because `std::function::operator()` does not document itself as signal safe, which makes it UB to call per the rules of [[support.signal]](https://eel.is/c++draft/support.signal) (3.1).

Reply:
That's true, this seems vestigial at this point. Initially I was concerned that `sigsetjmp` would disrupt the cache enough we wouldn't be able to read the sidechannel, but testing doesn't seem to show that's the case. I'll remove this knob for now; we can add it back if need be.